examples/airflow-torch-training/airflow_example_torch_train.py

```diff
@@ -1,17 +1,117 @@
+# ```
+#
+# We'll be launching an AWS EC2 instance via [SkyPilot](https://github.com/skypilot-org/skypilot), so we need to
+# make sure our AWS credentials are set up:
+# ```shell
+# $ aws configure
+# $ sky check
+# ```
+#
+
+# Import some other libraries we need - namely Airflow, Runhouse, and a few others.
+import logging
 from datetime import datetime, timedelta
 
+import runhouse as rh
 from airflow import DAG
 from airflow.operators.bash_operator import BashOperator
 from airflow.operators.python import PythonOperator
 
-# Import the functions from the training script
-from callables import (
-    access_data_callable,
-    bring_up_cluster_callable,
-    down_cluster,
-    train_model_callable,
-)
+# ## Import the model class
+# This class is the same as the example in https://www.run.house/examples/torch-vision-mnist-basic-model-train-test
+from TorchBasicExample import DownloadData, SimpleTrainer
+
+logger = logging.getLogger(__name__)
+
+# ## Define the callable functions.
+# These will be called in sequence by the Airflow PythonOperator. Each task in the Airflow DAG becomes minimal.
+# These callables define both *what tasks are run* and also *where the tasks are run* - essentially, programatically controlling the dispatch.
+
+# We can bring up an on-demand cluster using Runhouse. You can access powerful usage patterns by defining compute in code. All subsequent steps connect to this cluster by name, but you can bring up other clusters for other steps.
+def bring_up_cluster_callable(**kwargs):
+    logger.info("Connecting to remote cluster")
+    cluster = rh.ondemand_cluster(
+        name="a10g-cluster", instance_type="A10G:1", provider="aws"
+    ).up_if_not()
+
+    print(cluster.is_up())
+    # cluster.save() ## Use if you have a Runhouse Den account to save and monitor the resource.
+
+
+# We will send the function to download data to the remote cluster and then invoke it to download the data to the remote machine. You can imagine that this is a data access or pre-processing step after which data is prepared.
+def access_data_callable(**kwargs):
+    logger.info("Step 2: Access data")
+    env = rh.env(name="test_env", reqs=["torch", "torchvision"])
+
+    cluster = rh.cluster(name="a10g-cluster").up_if_not()
+    remote_download = rh.function(DownloadData).to(cluster, env=env)
+    logger.info("Download function sent to remote")
+    remote_download()
+    logger.info("Downloaded")
+
+
+# Then we instantiate the trainer, and then invoke the training on the remote machine. On the remote, we have a GPU. This is also a natural point to split the workflow if we want to do some tasks on GPU and some on CPU.
+def train_model_callable(**kwargs):
+    cluster = rh.cluster(name="a10g-cluster").up_if_not()
+
+    env = rh.env(name="test_env", reqs=["torch", "torchvision"])
+
+    remote_torch_example = rh.module(SimpleTrainer).to(
+        cluster, env=env, name="torch-basic-training"
+    )
+
+    model = remote_torch_example()
```

Comment:
nit: Can we call this `remote_trainer` or something like that to be careful to distinguish it from the model proper (which lives inside the trainer)?